Week01/Problem02/ypang_02-func.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+def getFibonacci(maximum):
+    n = 2
+    maxlen = 1000
+    ans = np.empty(maxlen, dtype=int)
+    ans[0] = 1
+    ans[1] = 2
+
+    for i in range(2,maxlen):
+        ans[i] = ans[i-2] + ans[i-1]
+        if(ans[i] > maximum):
+            break
+        n += 1
+        
+    return ans[:n]
+
+
+def getEven(array):
+    n = 0
+    ans = np.empty(len(array), dtype=int)
+
+    for i in array:
+        if(i%2 == 0):
+            ans[n] = i
+            n += 1
+
+    return ans[:n]
+        
+
+
+
+"""MAIN"""
+
+fibSeq = getFibonacci(4000000)
```

Comment:
You may want to avoid making such large lists, if they get too big you can fill up your memory and crash your code.

Reply:
If you refer to the function getFibonacci(…) that I was calling, the array I created is only 1000 words long, which is not that bad. Beside, an alternative answer provided in ypang_02.py was provided for on-the-fly creation of Fibonacci series. The ypang_02-func.py was intended to be an extra solution for me to practice the use of functions only.